src/config.py

```diff
@@ -15,7 +15,8 @@ class Config:
 
     MONGOS_PORT = 27018
     MONGODB_PORT = 27017
-    SUBSTRATE = "vm"
+    VM_SUBSTRATE = "vm"
+    K8S_SUBSTRATE = "k8s"
```

Comment:
I'm not a huge fan of bringing `K8S_SUBSTRATE` and `VM_SUBSTRATE` like that in the config as it brings another layer of "knowledge" about that in the charm code that could more easily bring havoc in the charm (eg: "What if I start checking about my substrate in my own charm").

Maybe adding those two possibilities in the library instead could help event it out: the library has the control and all the knowledge about the substrate given what the charm provides, which makes it more easily usable: Here, this library update brings a change in the charm config which will have to be replicated in the k8s charm and forever both charms would hold this information about the other substrate.

So my suggestion would be:
 * Keep `SUBSTRATE` here as it is
 * Do an enum, or a tiny class in the library for those two values like so:
```python3
class Substrate:
   VM = "vm"
   K8S = "k8s"
```
In a perfect world we would have StrEnum but we're not in python 3.11 yet so we'll have to do with it.

Reply:
As a side note, we can have string enums in Python 3.10 by doing the following, but then it must always be cast to str:
```
class Substrate(str, Enum)
    VM = "vm"
    K8s = "k8s"

print(f"{Substrate.VM}")
```